doc/wait-for-ready.md

```diff
@@ -1,14 +1,26 @@
 gRPC Wait for Ready Semantics
 =============================
 
-If an RPC is issued but the channel is in `TRANSIENT_FAILURE` or `SHUTDOWN`
-states, the RPC is unable to be transmited promptly. By default, gRPC
-implementations SHOULD fail such RPCs immediately. This is known as "fail fast,"
-but usage of the term is historical. RPCs SHOULD NOT fail as a result of the
-channel being in other states (`CONNECTING`, `READY`, or `IDLE`).
-
-gRPC implementations MAY provide a per-RPC option to not fail RPCs as a result
-of the channel being in `TRANSIENT_FAILURE` state. Instead, the implementation
-queues the RPCs until the channel is `READY`. This is known as "wait for ready."
-The RPCs SHOULD still fail before `READY` if there are unrelated reasons, such
-as the channel is `SHUTDOWN` or the RPC's deadline is reached.
+Errors can occur in many places. For the propose of this document we can bucket the errors into three separate situations:
+
+1. The RPC never leaves the gRPC client library, for example:
+    * The connection was broken and the channel is in a `TRANSIENT_FAILURE` state.
```

Comment:
"The connection was broken" does not trigger "TRANSIENT_FAILURE state". "The connection was broken" triggers IDLE. It sounds like you just want "the channel is in a `TRANSIENT_FAILURE` state".

Reply:
Due to a know c core issue, a broken connection can lead to TRANSIENT_FAILURE in some cases. However, I realize this is an implementation detail, and should be removed from the doc. I have removed specific examples to more general: If an RPC never leaves the client, it belongs in this bucket.

Reply:
Also: I was wrong, TRANSIENT_FAILURE can be triggered by broken connections.